### PR TITLE
Added Con[V]ex Hull  and Shad[O]ws Rendering

### PR DESCRIPTION
**Pull request by rohit-kumar-j**
from [deepmind/mujoco/discussions/336](https://github.com/deepmind/mujoco/discussions/336)
Normal | Convex Hull
:-:|:-:
  |  

No Shadows | Shadows
:-:|:-:
 |

### CODE REVIEW

**Comment by rohanpsingh** [comment]
Hi @rohit-kumar-j 
Thanks for the PR.

This PR fails with the following:

> AttributeError: 'MujocoViewer' object has no attribute 'callbacks'

Is it because I didn't merge PR #5 yet?
If yes, could you please make it work with the main? Then, I'll make the changes for putting the callbacks in a separate file myself.

**Comment by rohit-kumar-j** [comment]
Made the changes. It should work now. Sorry about that. Callbacks were not ready yet.

File: mujoco_viewer/mujoco_viewer.py [hunk 1]
```diff
@@ -23,7 +23,9 @@ def __init__(self, model, data):
         self._transparent = False
         self._contacts = False
         self._joints = False
+        self._shadows = False
```
Comment (rohanpsingh) [inline mujoco_viewer/mujoco_viewer.py]: `self._shadows = True`
since shadows are ON by default

File: mujoco_viewer/mujoco_viewer.py [hunk 2]
```diff
@@ -137,6 +139,16 @@ def _key_callback(self, window, key, scancode, action, mods):
...
+        # Shadow Rendering
+        elif key == glfw.KEY_O:
+            self._shadows = not self._shadows
+            self.scn.flags[mujoco.mjtRndFlag.mjRND_SHADOW] = self._shadows
+        # Convex-Hull rendering
+        elif key == glfw.KEY_V:
+            self.vopt.flags[
+                mujoco.mjtVisFlag.mjVIS_CONVEXHULL
+            ] = self._convex_hull_rendering
+            self._convex_hull_rendering = not self._convex_hull_rendering
```
Comment (rohanpsingh) [inline mujoco_viewer/mujoco_viewer.py]: This line should come before setting the vopt flags

Comment (rohanpsingh) [reply mujoco_viewer/mujoco_viewer.py]: This is the correct order:
```
            self._convex_hull_rendering = not self._convex_hull_rendering
            self.vopt.flags[
                mujoco.mjtVisFlag.mjVIS_CONVEXHULL
            ] = self._convex_hull_rendering
```

Comment (rohit-kumar-j) [reply mujoco_viewer/mujoco_viewer.py]: updated: [`23c7cea`](https://github.com/rohanpsingh/mujoco-python-viewer/commit/23c7cea70f9095804301216c39dbadef9866e413)

File: mujoco_viewer/mujoco_viewer.py (hunk 1 above)
Comment (rohit-kumar-j) [reply mujoco_viewer/mujoco_viewer.py]: Updated: [f5a39c1](https://github.com/rohit-kumar-j/mujoco-python-viewer/commit/f5a39c1970e486984952fdc0e3897d0f2b57df0f)

**Pull request merged**